alacritty_terminal/src/grid/mod.rs

```diff
@@ -232,21 +221,12 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
     }
 
     fn increase_scroll_limit(&mut self, count: usize, template: &T) {
-        self.scroll_limit = min(self.scroll_limit + count, self.max_scroll_limit);
-
-        // Initialize new lines when the history buffer is smaller than the scroll limit
-        let history_size = self.raw.len().saturating_sub(*self.lines);
-        if history_size < self.scroll_limit {
-            let new = min(
-                max(self.scroll_limit - history_size, MIN_INIT_SIZE),
-                self.max_scroll_limit - history_size,
-            );
-            self.raw.initialize(new, Row::new(self.cols, template));
-        }
+        let new = min(count, self.max_scroll_limit - self.history_size());
+        self.raw.initialize(new, template, self.cols);
```

Comment:
Actually, I think like it'll be a good idea to add a check for zero and don't call this function to initialize zero rows, what do you think? I feel like this was the original issue about perf I've faced, since I forgot to add a proper check for zero here, and once you added a way to move row and do this logic on initialize you basically avoided it.